src/python/pants/engine/isolated_process.py

```diff
@@ -98,15 +98,17 @@ def _post_process_execution(output_conversion, *args):
 class Binary(object):
   """Binary in the product graph.
 
-  TODO these should use BinaryUtil to find binaries.
+  TODO(cosmicexplorer): this should probably be coupled with instances of
+  BinaryTool subclasses.
   """
 
   @abstractproperty
   def bin_path(self):
     pass
 
-  def prefix_of_command(self):
-    return tuple([self.bin_path])
+  @abstractmethod
+  def gen_argv(self, snapshots):
```

Comment:
Can you add some documentation on this method explaining what it does and, in particular, what `snapshots` means? I'm not convinced that "snapshots" is a general-purpose enough abstraction in an interface here... Maybe let's remove the abstract method for now, and use ad-hoc methods on the subclasses for now?

Reply:
I was going to delete this from the Binary class but forgot to leave a TODO, totally agree and will do.